### DISCUSSION TITLE
crawl wrap does not fit all tracks

### DISCUSSION THREAD

**Discussion by SutorSW** [bmcclintock/momentuHMM#162] [Q&A] [answered]
Hello all,

My name is Sean and I'm a student at Texas Tech University working on some tortoise movement and connectivity analyses. I am working with temporally irregular data (missing observations and irregular time stamps) collected from GPS loggers. I am attempting to regularize the data and impute missing locations for very small time gaps (< 1 hr), and so I have been using crawlWrap. This seems to be working very well at regularizing the data, but I am having an issue I have not encountered in any vignette walk-throughs and that I am struggling to resolve: crawlWrap is not fitting all my tracks (in the case below, it is predicting locations for 1193/1664 tracks). Any ideas what might be causing this? 

**crw_out <- crawlWrap(obsData = data_split, timeStep = "30 min", 
                     Time.name = "time", coord = c("x", "y"))

Fitting 1664 track(s) using crawl::crwMLE...
DONE

Predicting locations (and uncertainty) at 30 mins time steps for 1193 track(s) using crawl::crwPredict...** 

Note that the many tracks are because I have split tracks that are greater than 2 hours apart and discarded any tracks shorter than 2 hours. I am awaiting permission to provide a small sample of the data so that I may be able to provide reproducible code.

Thank you in advance,

Sean

**Comment by bmcclintock** [comment, marked as answer]
It's likely that the failed tracks aren't converging due to poor initial values (`theta`) and/or insufficient data. If you search [Discussions](https://github.com/bmcclintock/momentuHMM/discussions) for "[initial values](https://github.com/bmcclintock/momentuHMM/discussions?discussions_q=initial+values)" or "[crawl](https://github.com/bmcclintock/momentuHMM/discussions?discussions_q=crawl)", you'll find tips for dealing with this. For example, see #79. See also the `theta` argument in the `crawlWrap` help file (`?crawlWrap`).

**Reply by SutorSW** [reply]
Thank you for getting back to me this. I had been giving "retryFits" a go, but specifying theta has certainly helped.
